src/otel.rs

```diff
@@ -0,0 +1,410 @@
+//! Support for OpenTelemetry.
+
+use std::sync::{Arc, LazyLock};
+
+use derive_where::derive_where;
+
+use opentelemetry::{
+    global::{BoxedTracer, ObjectSafeTracerProvider},
+    trace::{Span, SpanKind, TraceContextExt, Tracer, TracerProvider},
+    Context,
+    KeyValue,
+};
+
+use crate::{
+    bson::Bson,
+    cmap::{conn::wire::Message, Command, ConnectionInfo, StreamDescription},
+    error::{ErrorKind, Result},
+    operation::{aggregate::AggregateTarget, Operation},
+    options::{ClientOptions, ServerAddress, DEFAULT_PORT},
+    Client,
+    ClientSession,
+    Namespace,
+};
+
+#[cfg(test)]
+pub(crate) mod testing;
+
+/// Configuration for OpenTelemetry.
+#[derive(Clone, serde::Deserialize, typed_builder::TypedBuilder)]
+#[derive_where(Debug, PartialEq)]
+#[builder(field_defaults(default, setter(into)))]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
+pub struct Options {
```

Comment:
nit: I'd lean towards `OpentelemetryOptions` or `TracingOptions` here for consistency with other options structs; the generated docs also look a little odd with this name:

<img width="289" height="73" alt="Image" src="https://github.com/user-attachments/assets/f59eabac-8d7c-46c1-ad0b-d39c509aeb18" />

Reply:
Done.